MAINTAINERS.md

```diff
@@ -0,0 +1,18 @@
+# Maintainers
+
+This document aims to assist maintainers.
+
```

Comment:
Can you add some stuff on how to spot which changes are Nighthawk local vs. those that require deeper Envoy internals and architecture knowledge?

Reply:
done